### PR TITLE
chore: update changelog and bump version for 2.4.0

### PR DESCRIPTION
**Pull request by aaronchung-bitquill**
### By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,12 +3,17 @@ All notable changes to this project will be documented in this file.
 
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/), and this project adheres to [Semantic Versioning](https://semver.org/#semantic-versioning-200).
 
+## [2.4.0] - 2024-09-25
+
 ### :magic_wand: Added
 - Logic and a connection property to enable driver failover when network exceptions occur in the connect pipeline (PR #1099)[https://github.com/aws/aws-advanced-jdbc-wrapper/pull/1099]
 - A new reworked and re-architected failover plugin (PR #1089)[https://github.com/aws/aws-advanced-jdbc-wrapper/pull/1089]
 - Virtual Threading support (PR #1120)[https://github.com/aws/aws-advanced-jdbc-wrapper/pull/1120]
 - Limitless Connection Plugin. See [UsingTheLimitlessConnectionPlugin.md](https://github.com/aws/aws-advanced-jdbc-wrapper/blob/main/docs/using-the-jdbc-driver/using-plugins/UsingTheLimitlessConnectionPlugin.md).
```
Comment (karenc-bq) [inline CHANGELOG.md:12]:
```suggestion
- Logic and a connection property to enable driver failover when network exceptions occur in the connect pipeline ([PR #1099](https://github.com/aws/aws-advanced-jdbc-wrapper/pull/1099))
- A new reworked and re-architected failover plugin ([PR #1089](https://github.com/aws/aws-advanced-jdbc-wrapper/pull/1089))
- Virtual Threading support ([PR #1120](https://github.com/aws/aws-advanced-jdbc-wrapper/pull/1120))
- Limitless Connection Plugin. See [UsingTheLimitlessConnectionPlugin.md](https://github.com/aws/aws-advanced-jdbc-wrapper/blob/main/docs/using-the-jdbc-driver/using-plugins/UsingTheLimitlessConnectionPlugin.md).
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,12 +3,17 @@ All notable changes to this project will be documented in this file.
 
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/), and this project adheres to [Semantic Versioning](https://semver.org/#semantic-versioning-200).
 
+## [2.4.0] - 2024-09-25
+
 ### :magic_wand: Added
 - Logic and a connection property to enable driver failover when network exceptions occur in the connect pipeline (PR #1099)[https://github.com/aws/aws-advanced-jdbc-wrapper/pull/1099]
 - A new reworked and re-architected failover plugin (PR #1089)[https://github.com/aws/aws-advanced-jdbc-wrapper/pull/1089]
```
Comment (karenc-bq) [inline CHANGELOG.md:10]: It might be better to link to the failover 2 documentation instead of the PR.

File: CHANGELOG.md (hunk 1 above)
Comment (aaronchung-bitquill) [reply CHANGELOG.md:12]: Good eye!

Comment (karenc-bq) [inline CHANGELOG.md:12]: Maybe list Limitless Connection Plugin first.

**Review by karenc-bq — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (aaronchung-bitquill) [reply CHANGELOG.md]: Done!

Comment (aaronchung-bitquill) [reply CHANGELOG.md:9]: Done!

File: CHANGELOG.md (hunk 2 above)
Comment (aaronchung-bitquill) [reply CHANGELOG.md]: Done!

**Pull request merged**